src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -59,15 +62,30 @@ public Double quotaLimit(ClientQuotaType quotaType, Map<String, String> metricTa
         if (ClientQuotaType.PRODUCE.equals(quotaType) && currentStorageUsage > storageQuotaSoft && currentStorageUsage < storageQuotaHard) {
             double minThrottle = quotaMap.getOrDefault(quotaType, Quota.upperBound(Double.MAX_VALUE)).bound();
             double limit = minThrottle * (1.0 - (1.0 * (currentStorageUsage - storageQuotaSoft) / (storageQuotaHard - storageQuotaSoft)));
-            log.debug("Throttling producer rate because disk is beyond soft limit. Used: {}. Quota: {}", storageUsed, limit);
+            if (shouldLog(this.lastLoggedMessageSoftTimeMs)) {
+                log.debug("Throttling producer rate because disk is beyond soft limit. Used: {}. Quota: {}", storageUsed, limit);
+                this.lastLoggedMessageSoftTimeMs = System.currentTimeMillis();
+            }
             return limit;
         } else if (ClientQuotaType.PRODUCE.equals(quotaType) && currentStorageUsage >= storageQuotaHard) {
-            log.debug("Limiting producer rate because disk is full. Used: {}. Limit: {}", storageUsed, storageQuotaHard);
+            if (shouldLog(this.lastLoggedMessageHardTimeMs)) {
+                log.debug("Limiting producer rate because disk is full. Used: {}. Limit: {}", storageUsed, storageQuotaHard);
+                this.lastLoggedMessageHardTimeMs = System.currentTimeMillis();
+            }
             return 1.0;
         }
         return quotaMap.getOrDefault(quotaType, Quota.upperBound(Double.MAX_VALUE)).bound();
     }
 
+    /**
+     * Put a small delay between logging
+     * @param lastLoggedMessageTimeMs the last time the logger logged
+     * @return true when the logger is supposed to log again
+     */
+    private boolean shouldLog(long lastLoggedMessageTimeMs) {
+        return System.currentTimeMillis() - lastLoggedMessageTimeMs >= LOGGING_DELAY_MS;
```

Comment:
It's a bit ugly to have this method do the check on `lastLoggedMessageTimeMs` but have the caller update it. I think it would be better to write a `maybeLog(String format, Object... args)` method which encapsulated the read and write to `lastLoggedMessageTimeMs` and called `log.debug()`.